Feat/burger menu with breadcrumbs

VancheeZze opened the pull request:
<!--
  Before issuing a pull request:

  - Please read our contribution guidelines (https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/packages/bodiless-documentation/doc/Development/Contributing.md), especially the section on Pull Requests.
  - Please first create an issue (https://github.com/johnsonandjohnson/Bodiless-JS/issues/new). All pull requests must be linked to an issue.
-->

<!--
  IMPORTANT
  
  The pull request title will become the commit message title at merge, and
  should adhere to Angular Commit Message Conventions. Please see (https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/packages/bodiless-documentation/doc/Development/Contributing.md) for details and examples.
-->

## Changes
<!-- Brief description of the changes introduced by this PR -->

## Test Instructions
<!-- Detailed description of how this feature was (and should be) tested
  - Setup instructions (if any)?
  - List of test cases, with steps and expected results?
  - List
-->

## Related Issues
<!--
  Link to the issue that is fixed or resolved by this PR (if there is one)
  e.g. Fixes #1234, Closes #4567

  Link to an issue that is partially addressed by this PR (if there are any)
-->

<!-- IF THIS PR INTRODUCES A BREAKING CHANGE

BREAKING CHANGE: Describe the nature of the breaking change here.

More Details about the breaking change.
-->

Commit:
init commit for gtm interface

Commit:
add with GTM form

Commit:
updates to use meta pattern

Commit:
updates and fixes

Commit:
restore

Commit:
merge master

Commit:
bugfix

Commit:
add default as a prop

Commit:
updates

Commit:
fixes

Commit:
es lint fixes

Commit:
add rest

Commit:
aggregate implementation

Commit:
add tests

Commit:
remove console logs

Commit:
lint

Commit:
lint

Commit:
updates

Commit:
push items instead of array onto datalayer

Commit:
use ifeditable and refactor site level implementation to use withGlobalGTMForm hoc

Commit:
use ifeditable and refactor site level implementation to use withGlobalGTMForm hoc

Commit:
update product template

Commit:
ignore unsued import for now

Commit:
rm unused import

Commit:
cumelative withDefaultDatalayer

Commit:
Update gtm test

Commit:
use union for default datalayer aggregation

Commit:
add docs

Commit:
add docs

Commit:
update docs

:

Commit:
add docs

Commit:
add helmet to home page template

Commit:
remove old examples from starter

Commit:
Restore some files

Commit:
fix layout

Commit:
es lint

Commit:
use a named key on the datalayer object instead of array

Commit:
only display in edit mode

Commit:
fix lables

Commit:
fix types and test

Commit:
correct type

Commit:
correct type

Commit:
remove edit context from meta on site level

Commit:
address feedback

Commit:
remove ifeditable

Commit:
feedback: add ifEditable to withMetaForm

Commit:
merge upstream master and fix conflicts

Commit:
merge master & resolve conflicts

Commit:
fixes after merging master

Commit:
strong type withdataLayerItem to pass unit test

hvanyo committed:
Merge branch 'master' into feat/gtm-ui-aggregate

Commit:
address testing feedback

Commit:
address testing feedback

Commit:
Merge branch 'feat/gtm-ui-aggregate' of github.com:awm086/Bodiless-JS into feat/gtm-ui-aggregate

Commit:
add category to product

Commit:
move product helmet to a comp

Commit:
move product helmet to a comp

Commit:
address feedback

Commit:
address feedback

Commit:
feedback

Commit:
lint

Commit:
update empty values

Commit:
update empty values

Commit:
update empty values

Commit:
update empty values

Commit:
feedback

Commit:
feedback

Commit:
feedback

Commit:
Merge master into feat/gtm-ui-aggregat

Commit:
fix type error

Commit:
fix faling build by casting

Commit:
update type

Commit:
Implement Burger menu, update /megamenu/ page examples

Commit:
Fix typos

Commit:
Resolve merge conflicts

Commit:
Update top navigation menu

beliayeu committed:
pr review feedback

beliayeu committed:
pr feedback part 2

beliayeu committed:
jsdoc

beliayeu committed:
fixed ci build

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

Commit:
rc-menu cleanup

Commit:
rc-menu cleanup

Commit:
Minor Code cleanup

beliayeu committed:
withBreadcrumb -> withBreadcrumbs

Commit:
Merge branch 'master' into feat/gtm-ui-aggregate

Commit:
feat(components): Implement Interface for Managing GTM Datalayer

### Changes:
- Add an API to allow for composing user interface for GTM datalayers.

### Related Issues
Fixes #451

Commit:
Switch main nav with mega-menu

Commit:
Update Breadcrumbs

Commit:
Merge branch 'feat/breadcrumb-new' into feat/burger-menu-with-breadcrumbs

wodenx commented on packages/bodiless-organisms/src/components/Menu/BurgerMenu/asBurgerMenu.tsx:
```diff
@@ -0,0 +1,49 @@
+import { withMenuDesign as withSimpleMenuDesign } from '../SimpleMenu';
+import { withMenuDesign as withMegaMenuDesign } from '../MegaMenu';
+
+const withOverlayLinkAccordion = (Component: any) => {
```
Let's type this better.  Usually it's
```
const withX = <P extends object>(Component: ComponentType<P>) => {
  const WithX: FC<P> = {...}
  return WithX;
}
```

VancheeZze replied:
Updated

wodenx commented on packages/bodiless-organisms/src/components/Menu/BurgerMenu/asBurgerMenu.tsx:
```diff
@@ -0,0 +1,49 @@
+import { withMenuDesign as withMegaMenuDesign } from '../MegaMenu';
+
+const withOverlayLinkAccordion = (Component: any) => {
+  const FinalComponent: FC<any> = ({ children, overview, ...rest }) => (
```
name this WithOverlayLinkAccordion.  "FinalComponent" is not very helpful.

VancheeZze replied:
Updated

wodenx commented on packages/bodiless-components/src/List/asBodilessList.tsx:
```diff
@@ -45,26 +51,45 @@ const asTitledItem = <P extends TitledItemProps>(Item: ComponentType<P>) => {
 };
 
+const asOverviewItem = <P extends TitledItemProps>(Item: ComponentType<P>) => {
+  const ItemWithOverview: ComponentType<P> = (props) => {
```
this can be `FC<P>`

wodenx replied:
See my comment where this is referenced - i think we should move it there.

VancheeZze replied:
Done

wodenx commented on packages/bodiless-organisms/src/components/Menu/BurgerMenu/asBurgerMenu.tsx:
```diff
@@ -0,0 +1,49 @@
+    Title: asAccodionTitle,
+    WrapperItem: asAccordionWrapper,
+  }),
+  Item: asOverviewItem,
```
I think we should either bring asOverviewItem here, or move withOverlayLink (minus the accordion) into bodiless components.  Probably the former is better, as this is really burger menu specific behavior

wodenx replied:
and then maybe namespace that prop a bit - like "burgerMenuOverviewLink" or something

VancheeZze replied:
Done

dewen committed:
chore: update packages dependencies (#623)

* Update packages and dependencies for preparation for RT; Pre-1.0 manually updates; Slate v0.44.12;

Co-authored-by: Chris Oden <coden@its.jnj.com>
Co-authored-by: Dewen Li <dli94@its.jnj.com>

Commit:
Resolve merge conflicts

Commit:
Updata package-lock file

Commit:
Refactor asBurgerMenuOverviewLink

Commit:
Reuse tokens to style burger-menu, add missing navigation items

malachye committed:
docs (bodiless-documentation): Add content editor docs to YouTube and iframe components (#628)

* Update Youtube doc

- Edit and reformat YouTube component docs to add Content Editor/Site Builder/Architectural Details.

* Update iframe docs

-Edit/reformat iframe docs to include Content Editor/Site Builder/Architectural sections.

* Update Embed.md

-Reformat embed doc to add Site Builder and Architectural sections.

* Add screenshots for iframe and YouTube components

-Upload images for iframe and YouTube docs.

* Update YouTube.md

* Update defaultToc.ts

- Update TOC to correct the spelling of YouTube

- Fix spacing issue

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

Commit:
Disable accordion title link

Commit:
Update starter-site, fix overview link

Commit:
Fix starter

Commit:
Use SimpleMenu in starter site, update tokens

Commit:
Make Overview Link configurable

Commit:
Pass OverviewLink prop as a component

wodenx committed:
Add chameleon docs to toc. (#655)

doc(components): Add Chameleon docs to toc.

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

wodenx committed:
fix(layouts): Various FlowContainer bugs + Snap Indicator (#658)

## Changes
- Fix issues with flow container data (refresh and unnecessary save)
- Fix snap issues
- Add snap indicator so you'll know where you'll snap to when done resizing.
- Includes significant refactor of SortableChild

## Test Instructions
- See Linked Bugs

## Related Issues
Fixes #657 
Fixes #654 
Fixes #656

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

dewen committed:
feat(search): component library search component (#592)

* Component Library: Search Component

Co-authored-by: Dewen Li <dli94@its.jnj.com>

wodenx committed:
Spike/context menu groups (#597)

## Changes

- Provides customizable groups for context menu items.
- Revise labels of existing menu items

## Related Issues
- Closes #499

beliayeu committed:
fix: breadcrumb qa feedback #1 (#4)

* Enabled starting trail and final trail for breadcrumbs

* provided ability to override reducer that translates store items

* exposed configuration to enable/disable rendering last item as non-link

* updated snaphots

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

Commit:
Fix invalid export

Commit:
Update unit-test

beliayeu committed:
feat: Purge CSS Enhancements (#632)

* added package level css purging for bodiless/components and bodiless/organisms
* moved tailwindcss-aspect-ratio from site level to bodiless/organisms
* activated native tailwind css purging for test-site and starter
* removed gatsby-plugin-purgecss logic

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

Commit:
Add Burger Menu Documentation

Commit:
Code Cleanup, documentation update

beliayeu committed:
docs(bodiless-documentation): gatsby image configuration (#652)

* doc: gatsby image configuration

* Review feedback #1

Co-authored-by: malachy <30600153+malachye@users.noreply.github.com>

* Review feedback #2

Co-authored-by: malachy <30600153+malachye@users.noreply.github.com>

* Review feedback #3

Co-authored-by: malachy <30600153+malachye@users.noreply.github.com>

* moved image configuration to BuildingSites section

Co-authored-by: malachy <30600153+malachye@users.noreply.github.com>

dewen committed:
chore: Release v0.0.59 (#662)

* chore(release): publish v0.0.59

* Release v0.0.59 package.json and lock file updates.

* Trigger github check

* Trigger github check

* Trigger github check

Co-authored-by: Dewen Li <dli94@its.jnj.com>
Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>

marcopagliarulo committed:
fix(richtext): [GAPS] RTE editor doesn't allow for inline files/components (aka inline CTA styled buttons) (#481)

* fix(richtext): [GAPS] RTE editor doesn't allow for inline files/components (aka inline CTA styled buttons) #481

* fix(richtext): fix test errors #481

* fix(richtext): remove min-width workaround  #481

* fix(richtext): add style trick to prevent cursor moving on the right side of the content once focused #481

* fix(richtext): Add slate-react-placeholder declaration of and refactor the plugin usage to prevent performance issues #481

* fix(richtext): Prevent slate bug 1237 #481

* fix(richtext): apply patch for slate bug 1237 and improve inline style #481

* fix(richtext): update patch for slate bug 1237, fix root cause of cypress failures, remove workaround min width for menu items #481

* fix(richtext): PR review - remove ts-ignore #481

Co-authored-by: Marco Pagliarulo <mpaglia1@its.jnj.com>
Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>

wodenx committed:
fix(bodiless-layouts): Add local context menu and activator to flow container items. (#665)

Co-authored-by: hvanyo <58534559+hvanyo@users.noreply.github.com>

Commit:
Add responsive classes to menu

Commit:
Merge branch 'feat/burger-menu-with-breadcrumbs' of https://github.com/VancheeZze/Bodiless-JS into feat/burger-menu-with-breadcrumbs

Commit:
Resolve merge conflicts

Commit:
Update starter

beliayeu committed:
chore: activate breadcrumbs for starter + qa feedback #2 (#5)

* activated breadcrumbs for starter

* styled last breadrumb item from the breadcrumb store in last boldable item case

* made starting trail icon as a link

* activated latest breadcrumbs changes to starter

* linting

* enhanced isCurrentPage breadcrumb item logic

* addClassesIf -> ifToggledOn

* added an example of breadcrumbs with hidden current page item on test-site

* qa feedback #3

* propagated latest changes from test-site to starter

* added a failed test case for 3-level column megamenu

* remediated the failed test

* hit breadcrumb store only the first time during breadcrumb item rendering

* made ssr hit breadcrumb store on render, csr use useLayoutEffect

* canUseDOM small fix

* Revert "canUseDOM small fix"

This reverts commit cd2c3ca203837948e4f180b2f045cf4e6b3836d5.

* Revert "made ssr hit breadcrumb store on render, csr use useLayoutEffect"

This reverts commit 8f17b204f4cc96a9b10c2021512386ec87dc32eb.

* useRef instead of useState for checking if the component is rendered first time; extend breadcrumb test cases on demo page

Commit:
Address QA Feedback + better accessibility support for menus

Commit:
Code cleanup

Commit:
Tailwind cleanup

Commit:
Move Menu specific tailwind configs into bodiless/organisms

marcopagliarulo committed:
fix(richtext): [GAP] 2 clicks are needed on a newly created menu item to open Local context menu #671

## Changes
Prevent the on change state handler to save the data when it is the initial value and it has not been changed.
Preventing the unnecessary write data resolved the double clicks required to focus a newly created menu item.
The default Value provided to the richtext editor is converted to a new object to prevent focus issues on multi instance slate when the default value is identical between instances.

## Test Instructions
//=== for a clean page===
Open /megamenu/ in Edit mode
Click a top menu item
### Expected Results:
Local Context Menu is open.

//=== for a page that has content in Menu===
Open /megamenu/ in Edit mode
Click a top menu item
Click "+Add"
Click on the newly created item
### Expected Results:
Local Context Menu is open.
## Related Issues
Fixes #671

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

VancheeZze committed:
feat(components, ui): Updates to the Main Menu Links (#672)

## Changes
- When a link value is removed, the field returns to the initial state.

## Test Instructions
See https://github.com/johnsonandjohnson/Bodiless-JS/issues/593

## Related Issues
Closes: https://github.com/johnsonandjohnson/Bodiless-JS/issues/593

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

beliayeu committed:
feat(migration-tool): migrate site skeleton (#678)

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

wodenx committed:
feat(fclasses): Design Keys (#685)

## Changes
- Add "data-bl-design-key" attribute to designable elements.
- Control presence of this attribute based on "withShowDesighKeys" hoc
- Enable/disable in gatsby-theme-bodiless

## Related Issues

Closes #660

Commit:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs

Commit:
Add Design Keys

Commit:
Remove burger-menu page

Commit:
Remove Burger Menu link from menu

Commit:
feat(core, layouts, layouts-ui):  Add "Clear" Functionality for Component Library Filters (#669)

## Changes:
- Add "Clear" Functionality for Component Library Filters

## Test Instructions:
- Go to flow-container.
- Filter component library by checking/unchecking items 
- Use the Clear button to uncheck all. 

## Related Issues:
- Closes #668

VancheeZze committed:
Merge branch 'master' into feat/burger-menu-with-breadcrumbs
